src/dmd/optimize.d

```diff
@@ -306,9 +306,9 @@ Expression Expression_optimize(Expression e, int result, bool keepLvalue)
             return expOptimize(e.e1, flags);
         }
 
-        bool binOptimize(BinExp e, int flags)
+        bool binOptimize(BinExp e, int flags, bool keepLhsLvalue = false)
```

Comment:
No default parameters ([rule 12](https://github.com/dlang/dmd/blob/master/CONTRIBUTING.md#dmd-best-practices))

Reply:
There are 14 other call sites in this file I'd rather not touch.

Reply:
Let's say it's okay since it's not a new func.